File: julia/hobbs4_1.jl

# Define the Poisson probability function
function poisson_probability(y, λ)
    return exp(-λ) * λ^y / factorial(y)
end

# Given data
y1 = 14
y2 = 34
λ_true = 23

# Calculate individual probabilities
prob_y1_given_λ = poisson_probability(y1, λ_true)
prob_y2_given_λ = poisson_probability(y2, λ_true)

# Calculate joint probability (assuming independence)
joint_probability = prob_y1_given_λ * prob_y2_given_λ

# Print results
println("Probability of observing $y1 tadpoles given λ = $λ_true: $prob_y1_given_λ")
println("Probability of observing $y2 tadpoles given λ = $λ_true: $prob_y2_given_λ")
println("Joint probability of the data conditional on λ: $joint_probability")

File: R/hobbs4_1.R

# Example 4.1 Hobbs 2015: Poisson Distribution

# Given values
lambda <- 23  # True average number of tadpoles per liter
y1 <- 14      # Number of tadpoles in the first sample
y2 <- 34      # Number of tadpoles in the second sample

# Calculate individual probabilities for each observation
prob_y1 <- dpois(y1, lambda)
prob_y2 <- dpois(y2, lambda)

# Calculate joint probability assuming independence
joint_prob <- prob_y1 * prob_y2

# Print results
cat("Probability of observing", y1, "tadpoles in the first sample:", prob_y1, "\n")
cat("Probability of observing", y2, "tadpoles in the second sample:", prob_y2, "\n")
cat("Joint probability of the data assuming independence:", joint_prob, "\n")

